purge-tool/src/main/java/dk/dbc/holdingsitems/purge/Purge.java

```diff
@@ -0,0 +1,199 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with opensearch.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package dk.dbc.holdingsitems.purge;
+
+import dk.dbc.holdingsitems.HoldingsItemsDAO;
+import dk.dbc.holdingsitems.HoldingsItemsException;
+import dk.dbc.holdingsitems.Record;
+import dk.dbc.holdingsitems.RecordCollection;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ */
+public class Purge
+{
+    private static final Logger log = LoggerFactory.getLogger( Purge.class );
+    private final String worker;
+    private final String agencyName;
+    private final int agencyId;
+
+    private final int commitEvery;
+    private final boolean dryRun;
+
+    private final Connection connection;
+    private final HoldingsItemsDAO dao;
+
+    /**
+     * Create the purge request
+     * @param connection DB connection
+     * @param dao data access object for database
+     * @param worker Name of worker to put in queue
+     * @param agencyName The name of the agency to verify against
+     * @param agencyId The agency ID to be processed 
+     * @param commitEvery Optionally commit in batches when purging large sets. 0 to disable option
+     * @param dryRun Optionally check but does not commit anything
+     */
+    public Purge(Connection connection, HoldingsItemsDAO dao, String worker, String agencyName, int agencyId, int commitEvery, boolean dryRun) {
+        log.debug("Purge for agency ID {} with Queue: '{}'", agencyId, worker);
+        this.connection = connection;
+        this.dao = dao;
+        this.worker = worker;
+        this.agencyName = agencyName;
+        this.agencyId = agencyId;
+        this.commitEvery = commitEvery;
+        this.dryRun = dryRun;
+    }
+
+    /**
+     * Process the purge
+     * @throws HoldingsItemsException if DAO threw error
+     * @throws SQLException in case of rollback or commit error
+     */
+    public void process() throws HoldingsItemsException, SQLException {
+        log.debug("Process");
+
+        long start = System.currentTimeMillis();
+        
+        Set<String> bibliographicIds = dao.getBibliographicIds(agencyId);
+        int recordsCount = bibliographicIds.size();
+        int purgeCount = 0;
+        log.info("Found {} Bibliographic Ids for agency {}", recordsCount, agencyId);
+        statusReport(bibliographicIds);
+
+        // Confirm agency to purge
+        log.info("Agency: {}, Name: '{}'", agencyId, agencyName);
+        
+        boolean acceptable = false;
+        while (!acceptable) {
```

Comment:
What if the agency does not have a name?
scanner.next() won't return without input

Reply:
Hvis agency ikke har en navn, så får man strengen "<unknown>" fra PurgeMain.
Men der skal laves en abort på tom streng